File: composeApp/src/androidMain/kotlin/co/jarias/flexapp/ui/screens/bingo/figure_selection/BingoFigureSelectionScreenState.kt
package co.jarias.flexapp.ui.screens.bingo.figure_selection

import co.jarias.flexapp.domain.BingoCell
import co.jarias.flexapp.domain.WinCondition

data class BingoFigureSelectionScreenState(
    val gameId: Long = 0,
    val cardGrid: List<List<BingoCell>> = emptyList(),
    val selectedFigure: WinCondition? = null,
    val customPattern: Set<Pair<Int, Int>> = emptySet(),
    val isLoading: Boolean = true,
    val isUpdating: Boolean = false,
    val errorMessage: String? = null,
    val gameReady: Boolean = false
)

File: iosApp/iosApp/Screens/bingo/figure_selection/BingoFigureSelectionScreenState.swift
import Foundation
import shared

struct BingoFigureSelectionScreenState {
    var gameId: Int64 = 0
    var cardGrid: [[BingoCell]] = []
    var isLoading: Bool = false
    var isUpdating: Bool = false
    var errorMessage: String? = nil
    var gameReady: Bool = false
    var selectedFigure: WinCondition? = nil
    var customPattern: Set<BingoCellPos> = []
}
